test/e2e/specs/site-editor/template-revert.spec.js

```diff
@@ -0,0 +1,272 @@
+		templateRevertUtils,
+	} ) => {
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		// Revert template and check state.
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+		const contentAfterSave = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).not.toEqual( contentAfterSave );
+
+		// Undo revert and check state again.
+		await page.click( 'role=button >> text="Undo"' );
+		const contentAfterUndo = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfterUndo );
+	} );
+
+	test( 'should show the edited content after revert and clicking undo in the notice', async ( {
+		page,
+		templateRevertUtils,
+	} ) => {
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		const snackBar = page.locator(
+			'role=button[name="Dismiss this notice"]'
+		);
+
+		await snackBar.locator( 'role=button >> text="Undo"' ).click();
+
+		const contentAfter = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfter );
+	} );
+
+	test( 'should show the original content after revert, clicking undo then redo in the header toolbar', async ( {
+		page,
+		templateRevertUtils,
+	} ) => {
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+		const headerToolbar = page.locator( 'role=region[name="Header"]' );
+		await headerToolbar.locator( '[aria-label="Undo"]' ).click();
+
+		const contentAfterUndo = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).not.toEqual( contentAfterUndo );
+
+		const redoButton = headerToolbar.locator( '[aria-label="Redo"]' );
+
+		await redoButton.click();
+
+		const contentAfterRedo = await templateRevertUtils.getCurrentSiteEditorContent();
+		expect( contentBefore ).toEqual( contentAfterRedo );
+	} );
+
+	test( 'should show the original content after revert, clicking undo in the notice then undo in the header toolbar', async ( {
+		page,
+		templateRevertUtils,
+	} ) => {
+		const contentBefore = await templateRevertUtils.getCurrentSiteEditorContent();
+
+		await templateRevertUtils.addDummyText();
+		await templateRevertUtils.save();
+		await templateRevertUtils.revertTemplate();
+		await templateRevertUtils.save();
+
+		// Click undo in the snackbar. This reverts revert template action.
+		const snackBar = page.locator(
+			'role=button[name="Dismiss this notice"]'
+		);
+		await snackBar.locator( 'role=button >> text="Undo"' ).click();
```

Comment:
Nit: We can chain selectors together since role selectors are often readable enough without having to assign intermediate variables.

```suggestion
		await page.click( 'role=button[name="Dismiss this notice"i] >> role=button >> text="Undo"' );
```

It's just a preference though, and your way is totally valid too.

Reply:
I hadn't realised this was possible! As per my comment above, chaining seems to be more reliable than splitting parents and children, so I'll do that everywhere.